File: native/android/app/src/main/java/com/mediconnect/nativeapp/AuthenticatorSetup.kt
package com.mediconnect.nativeapp

import com.amplifyframework.auth.result.step.AuthNextSignInStep
import com.amplifyframework.auth.result.step.AuthSignInStep

/** Memory-only enrollment material. Never serialize or put the key in diagnostic output. */
class AuthenticatorSetup private constructor(private val key: String) {
    fun displayKey(): String = key
    override fun toString(): String = "AuthenticatorSetup(redacted)"
    companion object {
        fun fromSdk(step: AuthNextSignInStep): AuthenticatorSetup? {
            if (step.signInStep != AuthSignInStep.CONTINUE_SIGN_IN_WITH_TOTP_SETUP) return null
            val key = step.totpSetupDetails?.sharedSecret?.takeUnless { it.isBlank() } ?: return null
            return AuthenticatorSetup(key)
        }
    }
}

File: native/ios/Sources/AuthenticatorSetup.swift
import Amplify
import Foundation

/// Memory-only enrollment material with redacted diagnostics; deliberately not Codable.
struct AuthenticatorSetup: Equatable, CustomStringConvertible, CustomDebugStringConvertible {
    private let key: String
    private init(key: String) { self.key = key }
    var displayKey: String { key }
    var description: String { "AuthenticatorSetup(redacted)" }
    var debugDescription: String { description }
    static func fromSDK(_ step: AuthSignInStep) -> AuthenticatorSetup? {
        guard case .continueSignInWithTOTPSetup(let details) = step,
              !details.sharedSecret.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else { return nil }
        return AuthenticatorSetup(key: details.sharedSecret)
    }
}
